server/auth/auth.go

```diff
@@ -24,18 +24,270 @@ import (
 	"go.etcd.io/etcd/clientv3"
 )
 
+const (
+	userPrefix = "users"
+	rolePrefix = "roles"
+)
+
 // RBACManager is used for the rbac storage, cache, management and enforcing logic.
 type RBACManager struct {
+	userManager
 	roleManager
 }
 
+// NewRBACManager creates a new RBACManager.
+func NewRBACManager(kv kv.Base) *RBACManager {
+	return &RBACManager{
+		userManager{
+			kv:    kv,
+			users: make(map[string]*User),
+		},
+		roleManager{
+			kv:    kv,
+			roles: make(map[string]*Role),
+		}}
+}
+
+type userManager struct {
+	kv    kv.Base
+	mu    sync.RWMutex
+	users map[string]*User
+}
+
+// newUserManager creates a new roleManager for debug purposes.
+func newUserManager(kv kv.Base) *userManager {
+	return &userManager{kv: kv, users: make(map[string]*User)}
+}
+
+// GetUser returns a user.
+func (m *userManager) GetUser(name string) (*User, error) {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+
+	user, ok := m.users[name]
+	if !ok {
+		return nil, errs.ErrUserNotFound.FastGenByArgs(name)
+	}
+
+	return user, nil
+}
+
+// GetUsers returns all available roles.
+func (m *userManager) GetUsers() map[string]*User {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+
+	return m.users
+}
+
+// CreateUser creates a new user.
+func (m *userManager) CreateUser(name string, password string) error {
+	_, err := m.GetUser(name)
+	if err == nil {
+		return errs.ErrUserExists.GenWithStackByArgs(name)
+	}
+
+	user, err := NewUser(name, GenerateHash(password))
+	if err != nil {
+		return err
+	}
+
+	userJSON, err := json.Marshal(user)
+	if err != nil {
+		return errs.ErrJSONMarshal.Wrap(err).GenWithStackByCause()
+	}
+	userPath := path.Join(userPrefix, name)
```

Comment:
I saw the use of this function for multi-times, maybe we can make it become a common method?

Reply:
Extracted into `GetUserPath` and `GetRolePath`.